### ISSUE TITLE
Advice on how to implement this (a little bit complex) network

### ISSUE THREAD

**Issue by hj1210** [fchollet/keras#3890]
I am trying to implement the following network in keras (this is an image from the caffe model for the same network)
![samplemodelcombination](https://cloud.githubusercontent.com/assets/6188832/18882572/ff4c00b2-84df-11e6-8019-6f89cf517912.jpg)
Here the layer, loss-img uses the loss based on two different middle layers without any output. I was thinking of an implementation in terms of multi-output network but couldn't find any way to make one of the middle layers as output for another one (or will just a hanging merge layer without any output work?) .
I can provide more details on the use case, if needed.

**Comment by keunwoochoi** [comment]
Have you checked out [`functional` API](https://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models)? Looks like it is not very different from the example in the documentation.

**Comment by hj1210** [comment]
Yes, I had read the documentation. The difference is in the second output layer (if i treat this network as multi-output network), since the layer 'loss-img' just minimizes the loss between layers 'fc3-imfeat' and 'small-1' without any pre-defined output.

**Comment by dieuwkehupkes** [comment]
How should the loss of loss-img be computed?

**Comment by hj1210** [comment]
The loss for image is the euclidean loss between the layer small-1 and fc3-imfeat.

**Comment by dieuwkehupkes** [comment]
Ahh I see. This is not really a very clean option, but I guess one solution would be to add a layer that is connected to small-1 and fc3-imfeat that has the identity matrix as weightmatrix for the connections to fc3-imfeat and and - the identity to small1 (or the other way around). This layer would then have values 1[i]-fc3-imfeat[i]. Then if you minimize sum squared error for that layer, this should have the required effect right?

**Comment by hj1210** [comment]
ah yes, seems like a good idea. Will try it and post the results here. Thanks!

**Comment by hj1210** [comment]
I was able to train the network using the suggestion above. Closing the issue. Thanks!
